### PR TITLE
Home 3.2 completed

### PR DESCRIPTION
**Pull request by YaroslavKSE**
Completed homework 3.2 Translated c++ code into swift + added new functionality for calculation of the high of the triangle for each of the side. Covered the code with tests. 
image

### CODE REVIEW

File: Home/Home 3.2/Triangle.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    func area() -> Double {
+        let s = (a + b + c) / 2
+        return sqrt(s * (s - a) * (s - b) * (s - c))
+    }
+    
+    func height(for side: Character) -> Double? {
+        let area = self.area()
+        switch side {
+        case "a":
```
Comment (vovaskochko) [inline Home/Home 3.2/Triangle.swift]: I hope it should be enum to strictly limit the possible inputs here

File: Home/Home 3.2/Triangle.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import Foundation
+
+class Triangle {
```
Comment (vovaskochko) [inline Home/Home 3.2/Triangle.swift:11]: The same comments related to input validation and so on

**Review by vovaskochko — CHANGES_REQUESTED** [review]

File: Home/Home 3.2/Triangle.swift (hunk 1 above)
Comment (YaroslavKSE) [reply Home/Home 3.2/Triangle.swift]: Refactored the code in [90ac7f0](https://github.com/YaroslavKSE/Professinal-Programming/pull/10/commits/90ac7f03ae60329ab88db27d18a6b1b2a16da546)

File: Home/Home 3.2/Triangle.swift (hunk 2 above)
Comment (YaroslavKSE) [reply Home/Home 3.2/Triangle.swift:11]: Refactored the code in [90ac7f0](https://github.com/YaroslavKSE/Professinal-Programming/pull/10/commits/90ac7f03ae60329ab88db27d18a6b1b2a16da546)

File: Home/Home 3.2/TriangleTests.swift [hunk 3]
```diff
@@ -4,85 +4,111 @@ class TriangleTests: XCTestCase {
...
-        let triangle = Triangle(a: 6, b: 8, c: 10)
+        let triangle = try! Triangle(a: 5, b: 5, c: 5)
         
         // Act
-        let result = triangle.area()
+        let result = triangle.calculateArea()
         
         // Assert
-        XCTAssertEqual(result, 24.0, accuracy: doubleAccuracy)
+        let expectedArea = (sqrt(3) / 4) * 25 // (sqrt(3) / 4) * a^2
```
Comment (vovaskochko) [inline Home/Home 3.2/TriangleTests.swift]: We can just calculate the value - it doesn't makes sense to have formula here - internal implementation can be changed but we expect that for given points some set of input produces some output.
Also formula may confuse readers.

File: Home/Home 3.2/TriangleTests.swift [hunk 4]
```diff
@@ -4,85 +4,111 @@ class TriangleTests: XCTestCase {
...
+        let triangle = try! Triangle(a: 5, b: 5, c: 8)
         
         // Act
-        let result = triangle.area()
+        let result = triangle.calculateArea()
         
         // Assert
-        XCTAssertEqual(result, 30.0, accuracy: doubleAccuracy)
+        let s = (5 + 5 + 8) / 2.0
+        let expectedArea = sqrt(s * (s - 5) * (s - 5) * (s - 8))
```
Comment (vovaskochko) [inline Home/Home 3.2/TriangleTests.swift]: the same comment - just calculate some value - as we should not copy paste logic from implementations across the code.

**Pull request merged**